app/assets/javascripts/discourse-common/addon/lib/deprecated.js

```diff
@@ -39,7 +39,7 @@ export default function deprecated(msg, options = {}) {
   msg = msg.join(" ");
 
   let consolePrefix = "";
-  if (window.Discourse) {
+  if (require.has("discourse/lib/source-identifier")) {
```

Comment:
`window.Discourse` isn't set in tests. Checking for this module is a much safer way to identify whether we're running outside of a normal app environment.